compositor_pipeline/src/queue.rs

```diff
@@ -1,38 +1,32 @@
 mod audio_queue;
 mod audio_queue_thread;
+mod utils;
 mod video_queue;
 mod video_queue_thread;
 
 use std::{
     sync::{Arc, Mutex},
-    thread,
     time::{Duration, Instant},
 };
 
 use compositor_render::{
-    error::ErrorStack, AudioSamples, AudioSamplesBatch, AudioSamplesSet, Frame, FrameSet,
-    Framerate, InputId,
+    AudioSamples, AudioSamplesBatch, AudioSamplesSet, Frame, FrameSet, Framerate, InputId,
 };
 use crossbeam_channel::{unbounded, Receiver, Sender};
 use log::error;
 use thiserror::Error;
 
-use crate::pipeline::AudioChannels;
+use crate::pipeline::{decoder::DecodedDataReceiver, AudioChannels};
 
 use self::{
     audio_queue::AudioQueue, audio_queue_thread::AudioQueueThread, video_queue::VideoQueue,
     video_queue_thread::VideoQueueThread,
 };
 
-pub struct InputType {
-    pub input_id: InputId,
-    pub video: Option<()>,
-    pub audio: Option<AudioOptions>,
-}
-
-pub struct AudioOptions {
-    pub sample_rate: u32,
-    pub channels: AudioChannels,
+#[derive(Debug, Clone)]
+pub struct InputOptions {
+    pub video: Option<Receiver<Frame>>,
+    pub audio: Option<Receiver<AudioSamplesBatch>>,
```

Comment:
Is this used anywhere?

Reply:
Nope, I missed that.
It looks like I defined type like that twice and I'm using the other one.